File: 算法/李明杰数据结构/算法刷题/链表/ListNode.java
package 算法刷题.链表;

public class ListNode {
    public int val;
    public ListNode next;

    public ListNode(int x) {
        val = x;
    }

    @Override
    public String toString() {
        if (val == 0) {
            return null;
        } else {
            return " " + val + " ->" + next;
        }
    }
}

File: 算法/js算法刷题/链表/ListNode.js
function printListNode(head) {
    let ret = "";
    while (head) {
        ret += head.val;
        if (head.next) ret += " => ";
        head = head.next;
    }
    return ret;
}

function ListNode(val) {
    this.val = val;
    this.next = null;
}
const A = new ListNode("A");
const B = new ListNode("B");
const C = new ListNode("C");
const D = new ListNode("D");
const E = new ListNode("E");
A.next = B;
B.next = C;
C.next = D;
D.next = E;
module.exports = {
    ListNode,
    printListNode,
    A
};